bin/hubot.js

```diff
@@ -89,121 +89,121 @@ if (options.create) {
-
-if (options.version) {
-  console.log(robot.version)
-  process.exit(0)
-}
-
-if (options.configCheck) {
-  loadScripts()
-  console.log('OK')
-  process.exit(0)
-}
+  if (options.configCheck) {
+    loadScripts()
+    console.log('OK')
+    process.exit(0)
+  }
 
-robot.adapter.once('connected', loadScripts)
+  robot.adapter.once('connected', () => loadScripts().then().catch(err => console.error(err)))
 
-robot.run()
+  robot.run()
 
-function loadScripts () {
-  robot.load(pathResolve('.', 'scripts'))
-  robot.load(pathResolve('.', 'src', 'scripts'))
+  async function loadScripts () {
+    await robot.load(pathResolve('.', 'scripts'))
+    await robot.load(pathResolve('.', 'src', 'scripts'))
 
-  loadHubotScripts()
-  loadExternalScripts()
+    loadHubotScripts()
+    loadExternalScripts()
 
-  options.scripts.forEach((scriptPath) => {
-    if (scriptPath[0] === '/') {
-      return robot.load(scriptPath)
+    for await (const scriptPath of options.scripts) {
+      if (scriptPath[0] === '/') {
+        return await robot.load(scriptPath)
+      }
+      await robot.load(pathResolve('.', scriptPath))
     }
+  }
 
-    robot.load(pathResolve('.', scriptPath))
-  })
-}
+  function loadHubotScripts () {
+    const hubotScripts = pathResolve('.', 'hubot-scripts.json')
+    let scripts
+    let scriptsPath
 
-function loadHubotScripts () {
-  const hubotScripts = pathResolve('.', 'hubot-scripts.json')
-  let scripts
-  let scriptsPath
+    if (fs.existsSync(hubotScripts)) {
+      let hubotScriptsWarning
+      const data = fs.readFileSync(hubotScripts)
 
-  if (fs.existsSync(hubotScripts)) {
-    let hubotScriptsWarning
-    const data = fs.readFileSync(hubotScripts)
+      if (data.length === 0) {
+        return
+      }
 
-    if (data.length === 0) {
-      return
-    }
+      try {
+        scripts = JSON.parse(data)
+        scriptsPath = pathResolve('node_modules', 'hubot-scripts', 'src', 'scripts')
+        robot.loadHubotScripts(scriptsPath, scripts)
+      } catch (error) {
+        const err = error
+        robot.logger.error(`Error parsing JSON data from hubot-scripts.json: ${err}`)
+        process.exit(1)
+      }
 
-    try {
-      scripts = JSON.parse(data)
-      scriptsPath = pathResolve('node_modules', 'hubot-scripts', 'src', 'scripts')
-      robot.loadHubotScripts(scriptsPath, scripts)
-    } catch (error) {
-      const err = error
-      robot.logger.error(`Error parsing JSON data from hubot-scripts.json: ${err}`)
-      process.exit(1)
-    }
+      hubotScriptsWarning = 'Loading scripts from hubot-scripts.json is deprecated and ' + 'will be removed in 3.0 (https://github.com/github/hubot-scripts/issues/1113) ' + 'in favor of packages for each script.\n\n'
 
-    hubotScriptsWarning = 'Loading scripts from hubot-scripts.json is deprecated and ' + 'will be removed in 3.0 (https://github.com/github/hubot-scripts/issues/1113) ' + 'in favor of packages for each script.\n\n'
```

Comment:
Not for this PR, but noting since the line is changed here: this deprecated behaviour should become an error now we've passed 3.0?